prediction_market_agent/agents/microchain_agent/microchain_agent.py

```diff
@@ -29,14 +40,41 @@
 from prediction_market_agent.utils import APIKeys
 
 
+def build_wxdai_functions(keys: APIKeys) -> list[Function]:
+    functions = []
+
+    wrapped_xdai = WrappedxDaiContract()
```

Comment:
This feels a bit off, because `WrappedxDaiContract` is already a class with contract function implementations, why build them again, but with LLMs?

I think this method could be easily defined as 

```python
def build_functions_from_contract(address: ChecksumAddress) -> list[Function]:
    ...
```

and then, we could use it for whatever contract we want, wdyt?

Reply:
like it, done